src/backend/executor/nodeModifyTable.c

```diff
@@ -3809,6 +3809,12 @@ ExecModifyTable(PlanState *pstate)
 		if (pstate->ps_ExprContext)
 			ResetExprContext(pstate->ps_ExprContext);
 
+		if (estate->es_processed == estate->tsql_row_count && estate->tsql_row_count > 0)
```

Comment:
We can replace this with  a hook. Something like:
```
if (check_rowcount_hook)
{
    int tsql_rowcount = (*check_rowcount_hook) ();
    if (estate->es_processed == tsql_rowcount && tsql_rowcount > 0)
    {  
         slot  = NULL;
         break;
    }
}
```

Then we will not have to add `int tsql_row_count` to the EState struct.

Reply:
Here `check_rowcount_hook` will call a function that simply returns `pltsql_rowcount`.